TodoList/Src/TodoList.Contracts/DI/DependencyResolutionFailedException.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace TodoList.Contracts.DI
+{
+    public class DependencyResolutionFailedException : Exception
+    {
+        public string RequiredTypeName { get; }
+
+        public DependencyResolutionFailedException(string message, string requiredTypeName, Exception innerException)
```

Comment:
Preferably require a more complex type and extract vital information from it:
```suggestion
        public DependencyResolutionFailedException(string message, Type unresolvedType, Exception innerException)
```

PS: Note that I renamed the parameter slightly since I believe that requiring is a feature of (in this case method's) declaration rather than something related to the actual value.